src/Compilers/Core/CommandLine/BuildProtocol.cs

```diff
@@ -72,9 +72,9 @@ internal class BuildRequest
 
         public static BuildRequest Create(RequestLanguage language,
                                           IList<string> args,
-                                          string? workingDirectory = null,
-                                          string? tempDirectory = null,
-                                          string? compilerHash = null,
+                                          string workingDirectory,
+                                          string tempDirectory,
+                                          string compilerHash,
```

Comment:
These had been previously annotated as allowing `null`  but upon further inspection it's clear that `null` is not allowed as a value here. The client never provides `null` and the server actively rejects requests that provide `null` here.